### PR TITLE
Fix va-table screen reader not announcing sort

### PR DESCRIPTION
**Pull request by ataker**
## Chromatic

https://va-table-sort-screen-reader-fix--65a6e2ed2314f7b8f98609d8.chromatic.com

## Description
Closes [Table sorting - screen reader not announcing on sort #3538](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/3538)

Solution I ended up on is just delaying the screen reader text update by 100ms.

Solutions that helped but didn't completely fix the problem:

- Moving screen reader text from `va-table-inner` to `va-table`. `aria=live="polite"` only announces on update, but `va-table-inner` is completely replaced every time a sort occurs, so no announcement occurred. This change helped a little, but consistently no announcement occurred in Safari.
- Only replace the body of the table, not the headers. This required a major rewrite of the sort code and helped, but the announcement still seemed to compete with the button announcement and would be missed.
- Updating screen reader text at different points in the lifecycle. None seems to have a noticeable effect.

## QA Checklist
- [x] Component maintains 1:1 parity with design mocks
- [x] Text is consistent with what's been provided in the mocks
- [x] Component behaves as expected across breakpoints
- [x] Accessibility expert has signed off on code changes (if applicable. If not applicable provide reason why)
- [ ] Designer has signed off on changes (if applicable. If not applicable provide reason why)
- [x] Tab order and focus state work as expected
- [ ] Changes have been tested against screen readers (if applicable. If not applicable provide reason why)
- [ ] New components are covered by e2e tests; updates to existing components are covered by existing test suite
- [ ] Changes have been tested in vets-website using Verdaccio (if applicable. If not applicable provide reason why)

## Screenshots
Screenshot 2024-12-13 at 11 22 32

## Acceptance criteria
- [ ] QA checklist has been completed
- [ ] Screenshots have been attached that cover desktop and mobile screens

## Definition of done
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)

### CODE REVIEW

**Review by Andrew565 — APPROVED** [review]
Clever solution!

**Comment by rsmithadhoc** [comment]
I'm currently doing testing on this. I think it may work, but I was getting some inconsistency with getting it to read out on Safari. I believe it is due to Safari doing things a bit slower than Chrome. I'll try a few things on my end and test it on Windows as well.

File: packages/web-components/src/components/va-table/va-table-inner/va-table-inner.tsx [hunk 1]
```diff
@@ -319,7 +319,9 @@ export class VaTableInner {
           this.updateSpan(th, thSorted, nextSortDirection, content);
 
           // update sr text to reflect sort
-          this.updateSRtext(thSorted, content, currentSortDirection);
+          setTimeout(()=>{
+            this.updateSRtext(thSorted, content, currentSortDirection);
+          },100)
```
Comment (rsmithadhoc) [inline packages/web-components/src/components/va-table/va-table-inner/va-table-inner.tsx:324]: Update to 500

**Review by rsmithadhoc — CHANGES_REQUESTED** [review]
@ataker The solution looks good, thanks for digging into this one. I'm proposing a slight tweak, but let me know if you see any issues with that.

I tested this with slow/fast voice in VoiceOver and a larger table.
- Voice speed didn't seem to affect anything.
- I found that as the table got larger, the timeout needed to be increased. The original 100ms with the current Storybook example didn't always work. But once I increased that, it worked 100% of the time.
    - 500ms supports 100 rows, so I think that is sufficient for now. It shouldn't impact the user experience, since there are things that need to be read out prior to reading out the SR text, which 500ms should be plenty. I think supporting 100 rows is sufficient, but if not we may need to look at only running `updateSRtext` after sorting has fully completed in a future update.
- I also tested in Chrome/JAWS and Firefox/NVDA and it announced as expected there.

Thanks!

File: packages/web-components/src/components/va-table/va-table-inner/va-table-inner.tsx (hunk 1 above)
Comment (ataker) [reply packages/web-components/src/components/va-table/va-table-inner/va-table-inner.tsx]: Thanks Ryan! Updated.

**Review by rsmithadhoc — APPROVED** [review]

**Pull request merged**